scripts/setup_integration/nats_integration_env.sh

```diff
@@ -20,25 +20,35 @@ ACTION=$1
 
 start_podman () {
   podman pod create --replace --name vector-test-integration-nats -p 4222:4222
-  podman run -d --pod=vector-test-integration-nats  --name vector_nats \
-	 nats
+  podman run -d --pod=vector-test-integration-nats  --name vector_nats nats
+
+  podman pod create --replace --name vector-test-integration-nats-userpass -p 4223:4222
+  podman run -d --pod=vector-test-integration-nats-userpass  --name vector_nats_userpass nats \
+    --user missioncritical --pass hunter2
 }
 
 start_docker () {
   docker network create vector-test-integration-nats
-  docker run -d --network=vector-test-integration-nats -p 4222:4222 --name vector_nats \
-	 nats
+  docker network create vector-test-integration-nats-userpass
+  docker run -d --network=vector-test-integration-nats -p 4222:4222 --name vector_nats nats
+  docker run -d --network=vector-test-integration-nats-userpass -p 4223:4222 --name vector_nats_userpass nats \
+    --user missioncritical --pass hunter2
```

Comment:
Could these two instances live in the same pod?

Reply:
Is such a thing possible with vanilla Docker? Other than hacking on the image itself to run two NATS server instances, that is. I'm not against doing that, but right now the tests are running against the "pure" official NATS image.

Reply:
I think it should be. You can see what we do for Elasticsearch running both HTTP and HTTPS in one pod: https://github.com/timberio/vector/blob/master/scripts/setup_integration/elasticsearch_integration_env.sh

Reply:
Hmm, looks like that's still creating separate containers (each `docker run` command is a distinct container). But the Docker network can be shared, and the `podman pod create` can be shared, so I'll do that!

Reply:
Sorry for the misunderstanding. That was exactly what I was asking about.

Reply:
I've pushed a commit simplifying the Docker/podman setup.